src/Symfony/Component/Messenger/EventListener/SendFailedMessageToFailureTransportListener.php

```diff
@@ -53,27 +49,26 @@ public function onMessageFailed(WorkerMessageFailedEvent $event)
             return;
         }
 
-        // remove the received stamp so it's redelivered
         $throwable = $event->getThrowable();
         if ($throwable instanceof HandlerFailedException) {
             $throwable = $throwable->getNestedExceptions()[0];
         }
 
         $flattenedException = class_exists(FlattenException::class) ? FlattenException::createFromThrowable($throwable) : null;
-        $envelope = $envelope->withoutAll(ReceivedStamp::class)
-            ->withoutAll(TransportMessageIdStamp::class)
```

Comment:
Why remove this? `TransportMessageIdStamp` is added by any senders/receivers with such a concept and is used in the failure transport commands. I think I originally removed it because I know (for example, in the case of the Doctrine transport) that a new id will be assigned when it's redelivered to the failure transport.

More broadly, perhaps this stamp should simply have the `NonSendableStampInterface`? It's something that the sender/receiver *adds* when it sends/receives... and so I can't think of a reason you would want to *deliver* this ever to the transport.

Reply:
We could make it non sendable. But it could be useful for debugging. See https://github.com/symfony/symfony/pull/32078#discussion_r294595839
I don't mind either way. But keeping it for retry but removing it for failure was inconsistent.